python-lib/dku_email_client.py

```diff
@@ -169,10 +172,16 @@ def send_single_email(self, sender, recipients, cc_recipients, bcc_recipients, e
         for mime_app in attachment_mimes:
             msg.attach(mime_app)
 
-        self.smtp.sendmail(sender, all_recipients, msg.as_string())
+        try:
+            self.smtp.sendmail(sender, all_recipients, msg.as_string())
+        finally:
+            # Explicitly reset the session or "Error: nested MAIL command" error is possible
+            self.smtp.rset()
```

Comment:
Explaining patch changes: with multiple recipients on the STMP client, with one failure the client would end up in a bad state and everything would fail. This problem would have come in when the multiple recipients feature was added (was not caused by the changes in this PR AFAIK)